clients/src/main/java/org/apache/kafka/common/requests/DeleteGroupsResponse.java

```diff
@@ -16,121 +16,86 @@
-     * COORDINATOR_NOT_AVAILABLE(15)
-     * NOT_COORDINATOR (16)
-     * INVALID_GROUP_ID(24)
-     * GROUP_AUTHORIZATION_FAILED(30)
-     * NON_EMPTY_GROUP(68)
-     * GROUP_ID_NOT_FOUND(69)
-     */
-
-    private final Map<String, Errors> errors;
-    private final int throttleTimeMs;
-
-    public DeleteGroupsResponse(Map<String, Errors> errors) {
-        this(DEFAULT_THROTTLE_TIME, errors);
-    }
-
-    public DeleteGroupsResponse(int throttleTimeMs, Map<String, Errors> errors) {
-        this.throttleTimeMs = throttleTimeMs;
-        this.errors = errors;
+    public DeleteGroupsResponse(DeleteGroupsResponseData data) {
+        this.data = data;
     }
 
     public DeleteGroupsResponse(Struct struct) {
-        this.throttleTimeMs = struct.getOrElse(THROTTLE_TIME_MS, DEFAULT_THROTTLE_TIME);
-        Object[] groupErrorCodesStructs = struct.getArray(GROUP_ERROR_CODES_KEY_NAME);
-        Map<String, Errors> errors = new HashMap<>();
-        for (Object groupErrorCodeStructObj : groupErrorCodesStructs) {
-            Struct groupErrorCodeStruct = (Struct) groupErrorCodeStructObj;
-            String group = groupErrorCodeStruct.get(GROUP_ID);
-            Errors error = Errors.forCode(groupErrorCodeStruct.get(ERROR_CODE));
-            errors.put(group, error);
-        }
-
-        this.errors = errors;
+        short latestVersion = (short) (DeleteGroupsResponseData.SCHEMAS.length - 1);
+        this.data = new DeleteGroupsResponseData(struct, latestVersion);
     }
 
-    @Override
-    protected Struct toStruct(short version) {
-        Struct struct = new Struct(ApiKeys.DELETE_GROUPS.responseSchema(version));
-        struct.setIfExists(THROTTLE_TIME_MS, throttleTimeMs);
-        List<Struct> groupErrorCodeStructs = new ArrayList<>(errors.size());
-        for (Map.Entry<String, Errors> groupError : errors.entrySet()) {
-            Struct groupErrorCodeStruct = struct.instance(GROUP_ERROR_CODES_KEY_NAME);
-            groupErrorCodeStruct.set(GROUP_ID, groupError.getKey());
-            groupErrorCodeStruct.set(ERROR_CODE, groupError.getValue().code());
-            groupErrorCodeStructs.add(groupErrorCodeStruct);
-        }
-        struct.set(GROUP_ERROR_CODES_KEY_NAME, groupErrorCodeStructs.toArray());
-        return struct;
+    public DeleteGroupsResponse(Struct struct, short version) {
+        this.data = new DeleteGroupsResponseData(struct, version);
     }
 
     @Override
-    public int throttleTimeMs() {
-        return throttleTimeMs;
+    protected Struct toStruct(short version) {
+        return data.toStruct(version);
     }
 
     public Map<String, Errors> errors() {
-        return errors;
+        Map<String, Errors> errorMap = new HashMap<>();
+        for (DeletableGroupResult result : data.results()) {
+            errorMap.put(result.groupId(), Errors.forCode(result.errorCode()));
+        }
+        return errorMap;
     }
 
     public boolean hasError(String group) {
```

Comment:
Is this method used anywhere? The only caller I can find is `DeleteGroupsResponseTest` and that also test the Errors is None. 
If we decide to keep it, we can remove `.code()` from both sides of the equals

Reply:
Deleting this function